### PR TITLE
Fix LDAP and Kerberos config handling

### PR DESCRIPTION
**Pull request by traceon**
Changelog category:
- Not for changelog

Changelog entry:
- Do not allow LDAP server entries with the same name
- If Kerberos config has issues, disable it

Detailed description:
- Properly address #26197

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Should we add a test?

**Comment by traceon** [comment]
> Should we add a test?

@zvonand Could you please verify/add these cases to testflows?

File: src/Access/ExternalAuthenticators.cpp [hunk 1]
```diff
@@ -270,23 +270,37 @@ void ExternalAuthenticators::setConfiguration(const Poco::Util::AbstractConfigur
...
     config.keys("ldap_servers", ldap_server_names);
-    for (const auto & ldap_server_name : ldap_server_names)
+    ldap_client_params_blueprint.clear();
+    for (auto ldap_server_name : ldap_server_names)
     {
         try
         {
-            ldap_client_params_blueprint.erase(ldap_server_name);
-            parseLDAPServer(ldap_client_params_blueprint.emplace(ldap_server_name, LDAPClient::Params{}).first->second, config, ldap_server_name);
+            const auto bracket_pos = ldap_server_name.find('[');
```
Comment (nikitamikhaylov) [inline src/Access/ExternalAuthenticators.cpp:278]: How does `ldap_server_name` look like?  Why do you have to find that symbol and then resize the name...

Comment (traceon) [reply src/Access/ExternalAuthenticators.cpp:278]: For multiple XML tags with the same name on the same level, Poco's config parser will represent them as keys with `[1]`, `[2]`, etc., appended to the tag name.

See this, for example:
https://github.com/ClickHouse/ClickHouse/blob/1e5aaa4c10747163b00b20c54a1d4642b43651b5/src/Access/AccessControl.cpp#L325

Comment (traceon) [reply src/Access/ExternalAuthenticators.cpp:278]: I.e., for this:
```xml
<section>
    <subsection_A />
    <subsection_B />
    <subsection_A />
</section>
```
the keys would be names as: `subsection_A`, `subsection_B`, `subsection_A[2]`, or something like that.

File: src/Access/ExternalAuthenticators.cpp [hunk 2]
```diff
@@ -270,23 +270,37 @@ void ExternalAuthenticators::setConfiguration(const Poco::Util::AbstractConfigur
...
+            if (ldap_client_params_blueprint.count(ldap_server_name) > 0)
+                throw Exception("Multiple LDAP servers with the same name are not allowed", ErrorCodes::BAD_ARGUMENTS);
+
+            LDAPClient::Params ldap_client_params_tmp;
+            parseLDAPServer(ldap_client_params_tmp, config, ldap_server_name);
+            ldap_client_params_blueprint.emplace(std::move(ldap_server_name), std::move(ldap_client_params_tmp));
         }
         catch (...)
         {
             tryLogCurrentException(log, "Could not parse LDAP server " + backQuote(ldap_server_name));
```
Comment (alexey-milovidov) [inline src/Access/ExternalAuthenticators.cpp:291]: Why swallow exception? Let it unwind the stack...

File: src/Access/ExternalAuthenticators.cpp [hunk 3]
```diff
@@ -270,23 +270,37 @@ void ExternalAuthenticators::setConfiguration(const Poco::Util::AbstractConfigur
...
         if (kerberos_keys_count > 0)
-            parseKerberosParams(kerberos_params.emplace(), config);
+        {
+            GSSAcceptorContext::Params kerberos_params_tmp;
+            parseKerberosParams(kerberos_params_tmp, config);
+            kerberos_params = std::move(kerberos_params_tmp);
+        }
     }
     catch (...)
     {
```
Comment (alexey-milovidov) [inline src/Access/ExternalAuthenticators.cpp:306]: And here.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: src/Access/ExternalAuthenticators.cpp (hunk 2 above)
Comment (traceon) [reply src/Access/ExternalAuthenticators.cpp:291]: `tryLogCurrentException` should print the stack. Other than that, the error is _recoverable_, the rest can be parsed.

**Review by alexey-milovidov — APPROVED** [review]
Ok.

**Comment by alexey-milovidov** [comment]
> @vzakaznikov Could you please verify/add these cases to testflows?

Looks like Testflows are unmaintainable, let's merge without them.

**Pull request merged**